src/variables.jl

```diff
@@ -0,0 +1,259 @@
+                      name::String = "")::LogicalVariableRef
+                
+Extend `JuMP.add_variable` for [`LogicalVariable`](@ref)s. This 
+helps enable `@variable(model, [var_expr], LogicalVariable)`.
+"""
+function JuMP.add_variable(
+    model::JuMP.Model, 
+    v::LogicalVariable, 
+    name::String = ""
+    )
+    is_gdp_model(model) || error("Can only add logical variables to `GDPModel`s.")
+    data = LogicalVariableData(v, name)
+    idx = _MOIUC.add_item(_logical_variables(model), data)
+    _set_ready_to_optimize(model, false)
+    return LogicalVariableRef(model, idx)
+end
+
+# Base extensions
+Base.copy(v::LogicalVariableRef) = v
+Base.broadcastable(v::LogicalVariableRef) = Ref(v)
+Base.length(v::LogicalVariableRef) = 1
+function Base.:(==)(v::LogicalVariableRef, w::LogicalVariableRef)
+    return v.model === w.model && v.index == w.index
+end
+function Base.getindex(map::JuMP.ReferenceMap, vref::LogicalVariableRef)
+    return LogicalVariableRef(map.model, JuMP.index(vref))
+end
+
+# JuMP extensions
+JuMP.owner_model(vref::LogicalVariableRef) = vref.model
+JuMP.index(vref::LogicalVariableRef) = vref.index
+JuMP.isequal_canonical(v::LogicalVariableRef, w::LogicalVariableRef) = v == w
+function JuMP.is_valid(model::JuMP.Model, vref::LogicalVariableRef)
+    return model === JuMP.owner_model(vref)
+end
+
+function JuMP.name(vref::LogicalVariableRef)
+    data = gdp_data(JuMP.owner_model(vref))
+    return data.logical_variables[JuMP.index(vref)].name
+end
+function JuMP.set_name(vref::LogicalVariableRef, name::String)
+    data = gdp_data(JuMP.owner_model(vref))
+    data.logical_variables[JuMP.index(vref)].name = name
+    return
+end
+
+function JuMP.start_value(vref::LogicalVariableRef)
+    data = gdp_data(JuMP.owner_model(vref))
+    return data.logical_variables[JuMP.index(vref)].variable.start_value
+end
+function JuMP.set_start_value(
+    vref::LogicalVariableRef, 
+    value::Union{Nothing, Bool}
+    )
+    data = gdp_data(JuMP.owner_model(vref))
+    var = data.logical_variables[JuMP.index(vref)].variable
+    new_var = LogicalVariable(var.fix_value, value)
+    data.logical_variables[JuMP.index(vref)].variable = new_var
+    return
+end
+
+function JuMP.fix_value(vref::LogicalVariableRef)
+    data = gdp_data(JuMP.owner_model(vref))
+    return data.logical_variables[JuMP.index(vref)].variable.fix_value
+end
+function JuMP.fix(vref::LogicalVariableRef, value::Bool)
+    data = gdp_data(JuMP.owner_model(vref))
+    var = data.logical_variables[JuMP.index(vref)].variable
+    new_var = LogicalVariable(value, var.start_value)
+    data.logical_variables[JuMP.index(vref)].variable = new_var
+    return
+end
+function JuMP.unfix(vref::LogicalVariableRef)
+    data = gdp_data(JuMP.owner_model(vref))
+    var = data.logical_variables[JuMP.index(vref)].variable
+    new_var = LogicalVariable(nothing, var.start_value)
+    data.logical_variables[JuMP.index(vref)].variable = new_var
+    return
+end
```

Comment:
We should document these with docstrings to include in the docs.

Reply:
Also some of these should use `_set_ready_to_optimize(model, false)`

Reply:
done